src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py

```diff
@@ -44,6 +47,239 @@ class Bokeh(VisualizeLibInterface):
+                source=source,
+                color=color,
+                bottom=bottom,
+                # legend_label=y_label,
+            )
+            legend_manager.add_legend(y_label, renderer, y_label)
+            # color = next(color_generator)
+        color = color_selector.get_color(y_labels[-1])
+        renderer = fig.vbar(
+            x=x_label,
+            top=y_labels[-1],
+            width='x_width_list',
+            source=source,
+            color=color,
+            # legend_label=y_labels[-1],
+        )
+        legend_manager.add_legend(y_labels[-1], renderer, y_labels[-1])
+
+        num_legend_threshold = 20
+        legends = legend_manager.create_legends(num_legend_threshold, full_legends)
+        for legend in legends:
+            fig.add_layout(legend, 'right')
+        fig.legend.click_policy = 'mute'
+
+        return fig
+
+    def _get_stacked_bar_data(
+        self,
+        data: Dict[str, List[int]],
+        y_labels: List[str],
+        xaxis_type: str,
+        x_label: str,
+    ) -> Tuple[Dict[str, List[float]],  List[float]]:
+        """
+        Calculate stacked bar data.
+
+        Parameters
+        ----------
+        data : Dict[str, List[int]]
+            Source data.
+        y_labels : List[str]
+            Y axis labels that are Node/Topic name.
+        xaxis_type : str
+            Type of x-axis of the line graph to be plotted.
+            "system_time", "index", or "sim_time" can be specified.
+            The default is "system_time".
+        x_label : str
+            X-axis label of data dict.
+            "start time" or "index".
+
+        Returns
+        -------
+        Dict[str, List[float]]
+            Stacked bar data.
+        List[float]
+            Width list of bars.
+
+        """
+        output_data: Dict[str, List[float]] = {}
+        x_width_list: List[float] = []
+
+        # Convert the data unit to second
+        output_data = self._update_data_unit(data, 1e-9)
+
+        # Calculate the stacked y values
+        output_data = self._stacked_y_values(output_data, y_labels)
+
+        if xaxis_type == 'system_time':
+            # Update the timestamps from absolutely time to offset time
+            output_data[x_label] = self._update_timestamps_to_offset_time(output_data[x_label])
+
+            x_width_list = self._get_x_width_list(output_data[x_label])
+            half_width_list = [x / 2 for x in x_width_list]
+
+            # Slide x axis values so that the bottom left of bars are the start time.
+            output_data[x_label] = self._add_shift_value(output_data[x_label], half_width_list)
+        elif xaxis_type == 'sim_time':
+            raise NotImplementedError()
+        else:  # index
+            output_data[x_label] = list(range(0, len(output_data[y_labels[0]])))
+            x_width_list = self._get_x_width_list(output_data[x_label])
+
+        return output_data, x_width_list
+
+    @staticmethod
+    def _update_data_unit(
```

Comment:
This function appears to update `self` despite `static method`.

The function name may be changed.
e.g.
```suggestion
    def _data_on_unit(
```